yodeploy/application.py

```diff
@@ -105,15 +105,15 @@ def hook(self, hook, target, repository, version):
         if target:
             cmd += ['--target', target]
         cmd += [self.appdir, version]
-        try:
-            subprocess.check_call(cmd, env={
-                    'PATH': os.environ['PATH'],
-            })
-        except subprocess.CalledProcessError:
+
+        cmd_proc = subprocess.Popen(
+            cmd, env={'PATH': os.environ['PATH']}, close_fds=False)
+        cmd_proc.wait()
+        tlss.shutdown()
```

Comment:
You're totally right. Aside from the `close_fds` addition, this was a bad change. I'm going to mostly restore the previous behavior here.